roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python
+'''
+  Module for performing checks on an elasticseatch, fluentd, and kibana stack
+'''
+# pylint: disable=missing-docstring
+# vim: expandtab:tabstop=4:shiftwidth=4
+#
+#   Copyright 2017 Red Hat Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+import json
+import os
+import ssl
+import urllib2
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class EFKStackCheck(OpenShiftCheck):
+    """
+        This is a check for the entire EFK stack shipped with OCP
+        https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py
+    """
+
+    name = "efk_stack_check"
+    tags = ["elasticsearch", "efkstack"]
+
+    namespace = "logging"
+
+    es_pods = []
+    fluentd_pods = []
```

Comment:
It is poor coding style to store these lists as class attributes.
Think about it for a while ;)

The code becomes tied to calling a handful of methods in the right order to mutate / read from those...
It is very similar to global variables, misses encapsulation, promotes mutation of state, makes the whole thing harder to test and so on.

Instead, I suggest we return list of pods where they need to be returned, and pass explicitly as argument a list where there needs to be a list as input.